doubly_linked_list/doubly_linked_list.py

```diff
@@ -43,26 +43,74 @@ def __init__(self, node=None):
   def __len__(self):
     return self.length
 
+  """replaces the head of the list with a new value that 
+  is passed in.
+  """
   def add_to_head(self, value):
-    pass
+    new_node = ListNode(value)
+    if not self.head and not self.tail:
+      self.head = new_node
+      self.tail = new_node
+    else:
+      old_head = self.head
+      old_head.insert_before(value)
+      self.head = old_head.prev
 
+  """removes the head node and returns the value stored 
+  in it.
+  """
   def remove_from_head(self):
-    pass
+    if not self.head:
+      return None
+    else:
+      removed_head = self.head
+      self.head.delete()
+      return removed_head.value
 
+  """replaces the tail of the list with a new value that 
+  is passed in
+  """
   def add_to_tail(self, value):
-    pass
-
+    new_node = ListNode(value)
+    if not self.head and not self.tail:
+      self.head = new_node
+      self.tail = new_node
+    else:
```

Comment:
does this work? this seems like it would ... eat itself. I'll have to test that again.